src/support/hazard_weak.c

```diff
@@ -278,3 +264,73 @@ __wt_hazard_weak_invalidate(WT_SESSION_IMPL *session, WT_REF *ref)
     }
     WT_STAT_CONN_INCRV(session, cache_hazard_walks, walk_cnt);
 }
+
+/*
+ * __wt_hazard_weak_upgrade --
+ *     Attempts to convert a weak hazard pointer into a full hazard pointer, failing if it has been
+ *     invalidated.
+ */
+int
+__wt_hazard_weak_upgrade(WT_SESSION_IMPL *session, WT_HAZARD_WEAK **whpp, WT_REF **refp)
+{
+    WT_DECL_RET;
+    WT_HAZARD_WEAK *whp;
+    bool busy;
+
+    *refp = NULL;
+    whp = *whpp;
+
+    /* If a file can never be evicted, hazard pointers aren't required. */
+    if (F_ISSET(S2BT(session), WT_BTREE_IN_MEMORY))
+        return (0);
+
+    /*
+     * An empty slot reflects a serious error, we should always find the weak hazard pointer. Panic,
+     * because we messed up in and it could imply corruption.
+     */
+    if (whp == NULL || whp->ref == NULL)
+        WT_RET_PANIC(session, WT_PANIC,
+          "session %p: could not find the weak hazard pointer for a modify operation",
```

Comment:
This panic is a bit weird, we're passing in a weak hazard pointer that has been pulled of the txn's modification so it should be the responsibility of the caller to check if that weak hazard pointer isn't valid. Maybe this could be an assert instead?

Reply:
I am not hung up on using panic, replaced with an assert